src/systems/shadow-system.js

```diff
@@ -0,0 +1,73 @@
+import resizeShadowCameraFrustum from "../utils/resizeShadowCameraFrustum";
+
+function traverseAnimationTargets(rootObject, animations, callback) {
+  if (animations && animations.length > 0) {
+    for (const animation of animations) {
+      for (const track of animation.tracks) {
+        const { nodeName } = THREE.PropertyBinding.parseTrackName(track.name);
+        let animatedNode = rootObject.getObjectByProperty("uuid", nodeName);
+
+        if (!animatedNode) {
+          animatedNode = rootObject.getObjectByName(nodeName);
+        }
+
+        if (animatedNode) {
+          callback(animatedNode);
+        }
+      }
+    }
+  }
+}
+
+export class ShadowSystem {
+  constructor(sceneEl) {
+    this.needsUpdate = false;
+    this.dynamicShadowsEnabled = window.APP.store.state.preferences.enableDynamicShadows;
+    this.sceneEl = sceneEl;
+    this.onEnvironmentSceneLoaded = this.onEnvironmentSceneLoaded.bind(this);
+    this.sceneEl.addEventListener("environment-scene-loaded", this.onEnvironmentSceneLoaded);
+  }
+
+  onEnvironmentSceneLoaded({ detail: scene }) {
```

Comment:
Super minor nit, but I would call this `environment` or `environmentObject3D` so as not to confuse it with the aframe scene.